rfd/0184-agent-auto-updates.md

```diff
@@ -0,0 +1,1640 @@
+# dev          not started                             120                0                   0
+# staging      not started                             20                 0                   0
+# prod         not started                             234                0                   0
+```
+</details>
+
+I can trigger the dev group immediately using the command:
+
+```shell
+tctl autoupdate agent start-update dev --no-canary
+# Dev group update triggered (canary or active)
+```
+
+Alternatively
+```shell
+tctl autoupdate agent force-done dev
+```
+
+<details>
+<summary>After:</summary>
+
+```shell
+tctl autoupdate agent status
+# Rollout plan created the YYYY-MM-DD
+# Previous version: v2
+# New version: v3
+# Status: enabled
+# 
+# Group Name   Status              Update Start Time   Connected Agents   Up-to-date agents   failed updates
+# ----------   -----------------   -----------------   ----------------   -----------------   --------------
+# dev          not started                             120                0                   0
+# staging      not started                             20                 0                   0
+# prod         not started                             234                0                   0
+```
+</details>
+
+#### As a Teleport user, I want to install a new agent automatically updated
+
+The manual way:
+
+```bash
+wget https://cdn.teleport.dev/teleport-update-<os>-<arch>
+chmod +x teleport-update
+./teleport-update enable --proxy example.teleport.sh --group production
+# Detecting the Teleport version and edition used by cluster "example.teleport.sh"
+# Installing the following teleport version:
+#   Version: 16.2.1
+#   Edition: Enterprise
+#   OS: Linux
+#   Architecture: x86
+# Teleport installed
+# Enabling automatic updates, the agent is part of the "production" update group.
+# You can now configure the teleport agent with `teleport configure` or by writing your own `teleport.yaml`.
+# When the configuration is done, enable and start teleport by running:
+# `systemctl start teleport && systemctl enable teleport`
+```
+
+The one-liner:
+
+```
+curl https://cdn.teleport.dev/auto-install | bash -s example.teleport.sh
+# Downloading the teleport updater
+# Detecting the Teleport version and edition used by cluster "example.teleport.sh"
+# Installing the following teleport version:
+#   Version: 16.2.1
+#   Edition: Enterprise
+#   OS: Linux
+#   Architecture: x86
+# Teleport installed
+# Enabling automatic updates, the agent is part of the "default" update group.
+# You can now configure the teleport agent with `teleport configure` or by writing your own `teleport.yaml`.
+# When the configuration is finished, enable and start teleport by running:
+# `systemctl start teleport && systemctl enable teleport`
+```
+
+I can also install teleport using the package manager, then enroll the agent into AUs. See the section below:
+
+#### As a Teleport user I want to enroll my existing agent into AUs
```

Comment:
Does this also cover existing agents using our current automatic upgrade system?

Reply:
Yes, the UX should be the same. The updater might take extra steps to remove the old updater package from the system once everything is working.